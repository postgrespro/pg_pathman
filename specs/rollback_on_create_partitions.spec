setup
{
	CREATE EXTENSION pg_pathman;
	CREATE TABLE range_rel(id serial primary key);
}

teardown
{
	DROP TABLE range_rel CASCADE;
	DROP EXTENSION pg_pathman;
}

session "s1"
step "begin"				{ BEGIN; }
step "rollback"				{ ROLLBACK; }
step "commit"				{ COMMIT; }
step "insert_data"			{ INSERT INTO range_rel SELECT generate_series(1, 10000); }
step "create_partitions"	{ SELECT create_range_partitions('range_rel', 'id', 1, 1000); }
step "drop_partitions"		{ SELECT drop_partitions('range_rel'); }
step "savepoint_a"			{ SAVEPOINT a; }
step "rollback_a"			{ ROLLBACK TO SAVEPOINT a; }
step "savepoint_b"			{ SAVEPOINT b; }
step "rollback_b"			{ ROLLBACK TO SAVEPOINT b; }
step "savepoint_c"			{ SAVEPOINT c; }
step "show_rel"				{ SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass; }

permutation "begin" "insert_data" "create_partitions" "show_rel" "rollback" "show_rel"

permutation "begin" "insert_data" "create_partitions" "show_rel" "commit" "show_rel"

permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "rollback" "show_rel"
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "commit" "show_rel"

# rollback to 'b' after dropping partitions
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "savepoint_c" "rollback_b" "show_rel" "rollback" "show_rel"
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "savepoint_c" "rollback_b" "show_rel" "commit" "show_rel"

# rollback to 'a' after dropping partitions
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "rollback_a" "show_rel" "rollback" "show_rel"
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "rollback_a" "show_rel" "commit" "show_rel"

# drop partitions twice in a single transaction
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "rollback_b" "drop_partitions" "show_rel" "rollback" "show_rel"
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "show_rel" "savepoint_c" "rollback_b" "drop_partitions" "show_rel" "commit" "show_rel"

# create partitions twice in a single transaction
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "rollback_a" "create_partitions" "show_rel" "rollback" "show_rel"
permutation "begin" "insert_data" "savepoint_a" "create_partitions" "savepoint_b" "drop_partitions" "rollback_a" "create_partitions" "show_rel" "commit" "show_rel"
