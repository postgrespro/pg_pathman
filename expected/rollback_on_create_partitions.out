Parsed test spec with 1 sessions

starting permutation: begin insert_data create_partitions show_rel rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data create_partitions show_rel commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions savepoint_c rollback_b show_rel rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step savepoint_c: SAVEPOINT c;
step rollback_b: ROLLBACK TO SAVEPOINT b;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions savepoint_c rollback_b show_rel commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step savepoint_c: SAVEPOINT c;
step rollback_b: ROLLBACK TO SAVEPOINT b;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c rollback_a show_rel rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step rollback_a: ROLLBACK TO SAVEPOINT a;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c rollback_a show_rel commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step rollback_a: ROLLBACK TO SAVEPOINT a;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c rollback_b drop_partitions show_rel rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step rollback_b: ROLLBACK TO SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions show_rel savepoint_c rollback_b drop_partitions show_rel commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step savepoint_c: SAVEPOINT c;
step rollback_b: ROLLBACK TO SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)

step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions rollback_a create_partitions show_rel rollback show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step rollback_a: ROLLBACK TO SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step rollback: ROLLBACK;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent|partition
------+---------
(0 rows)


starting permutation: begin insert_data savepoint_a create_partitions savepoint_b drop_partitions rollback_a create_partitions show_rel commit show_rel
step begin: BEGIN;
step insert_data: INSERT INTO range_rel SELECT generate_series(1, 10000);
step savepoint_a: SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step savepoint_b: SAVEPOINT b;
s1: NOTICE:  1000 rows copied from range_rel_1
s1: NOTICE:  1000 rows copied from range_rel_2
s1: NOTICE:  1000 rows copied from range_rel_3
s1: NOTICE:  1000 rows copied from range_rel_4
s1: NOTICE:  1000 rows copied from range_rel_5
s1: NOTICE:  1000 rows copied from range_rel_6
s1: NOTICE:  1000 rows copied from range_rel_7
s1: NOTICE:  1000 rows copied from range_rel_8
s1: NOTICE:  1000 rows copied from range_rel_9
s1: NOTICE:  1000 rows copied from range_rel_10
step drop_partitions: SELECT drop_partitions('range_rel');
drop_partitions
---------------
             10
(1 row)

step rollback_a: ROLLBACK TO SAVEPOINT a;
step create_partitions: SELECT create_range_partitions('range_rel', 'id', 1, 1000);
create_range_partitions
-----------------------
                     10
(1 row)

step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

step commit: COMMIT;
step show_rel: SELECT l.parent, l.partition FROM pathman_partition_list l WHERE l.parent = 'range_rel'::regclass;
parent   |partition   
---------+------------
range_rel|range_rel_1 
range_rel|range_rel_2 
range_rel|range_rel_3 
range_rel|range_rel_4 
range_rel|range_rel_5 
range_rel|range_rel_6 
range_rel|range_rel_7 
range_rel|range_rel_8 
range_rel|range_rel_9 
range_rel|range_rel_10
(10 rows)

